setup.py

```diff
@@ -102,141 +100,42 @@ def run(self):
-                )
 
-                # If there's a mismatch, see what we can do.
-                if python_is_64 != cmake_is_64:
-                    if msvc_is_fixed_to_64 and not python_is_64:
-                        raise RuntimeError(
-                            "MSVC is configured to build 64-bit binaries, but Python is 32-bit!"
-                        )
-                    if not cmake_is_msvc:
-                        raise RuntimeError(
-                            "Mismatch in 32-bit/64-bit between CMake defaults ({}) and Python install ({})!".format(
-                                "64-bit" if cmake_is_64 else "32-bit",
-                                "64-bit" if python_is_64 else "32-bit",
-                            )
-                        )
-
-                    # Looks like we're compiling with MSVC, and MSVC is merely
-                    # defaulting to the wrong bitness, which means we should be
-                    # able to change it with the -A flag.
-                    if python_is_64:
-                        cmd = cmd["-A"]["x64"]
-                    else:
-                        cmd = cmd["-A"]["win32"]
-
-            # C++ tests can be enabled using an environment variable. They'll
-            # be run before the install.
-            if "QX_BUILD_TESTS" in os.environ:
-                cmd = cmd["-DQX_BUILD_TESTS=" + os.environ["QX_BUILD_TESTS"]]
-
-            # Enable CPU extension compatibility mode when requested.
-            if "QX_CPU_COMPATIBILITY_MODE" in os.environ:
-                cmd = cmd[
-                    "-DQX_CPU_COMPATIBILITY_MODE="
-                    + os.environ["QX_CPU_COMPATIBILITY_MODE"]
-                ]
+        # Configure and build using Conan
+        with local.cwd(root_dir):
+            build_type = os.environ.get("CMAKE_BUILD_TYPE", "Release")
+            build_tests = os.environ.get("QX_BUILD_TESTS", "False")
+            cpu_compatibility_mode = os.environ.get("QX_CPU_COMPATIBILITY_MODE", "False")
 
-            # Run cmake configuration.
+            cmd = local['conan']['profile']['detect']['--force']
             cmd & FG
 
-            # Do the build with the given number of parallel threads.
-            build_cmd = local["cmake"]["--build"]["."]["--config"][build_type]
-            cmd = build_cmd
-            if nprocs != "1":
-                try:
-                    parallel_supported = tuple(
-                        local["cmake"]("--version")
-                        .split("\n")[0]
-                        .split()[-1]
-                        .split(".")
-                    ) >= (3, 12)
-                except:
-                    parallel_supported = False
-                if parallel_supported:
-                    cmd = cmd["--parallel"][nprocs]
-                elif not sys.platform.startswith("win"):
-                    cmd = cmd["--"]["-j"][nprocs]
+            cmd = (local['conan']['create']['.']
+                ['--version']['0.6.5']
```

Comment:
We have to remember to update this version number.

This is something we will have to look into in the future. I don't like at all this hard coding of versions here and there. Maybe we could pass the version to `setup.py`, and, at the same time, automate that passing from GitHub, I don't really know. At the moment it is the way it is for QX simulator, libqasm...

Reply:
I've replaced this version number with a call to get_version() which parses the Version.hpp header to have a single source of truth